pkg/antctl/raw/migrate/convertnetworkpolicy/command.go

```diff
@@ -0,0 +1,488 @@
+			return err
+		}
+		if np.Spec.Order != nil {
+			anp.Spec.Priority = *np.Spec.Order
+		} else {
+			anp.Spec.Priority = defaultPriority
+		}
+		if _, err := c.antreaClient.CrdV1beta1().NetworkPolicies(np.Namespace).Create(context.Background(), &anp, metav1.CreateOptions{}); err != nil {
+			klog.Errorf("Failed to create Antrea NetworkPolicy %s: %v", np.Name, err)
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *converter) convertGlobalNetworkPolicy() error {
+	gnps, err := c.calicoClient.ProjectcalicoV3().GlobalNetworkPolicies().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("Failed to list GlobalNetworkPolicies: %v", err)
+		return err
+	}
+	for _, gnp := range gnps.Items {
+		klog.Infof("Converting GlobalNetworkPolicy %s", gnp.Name)
+		acnp := v1beta1.ClusterNetworkPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: gnp.Name,
+				Annotations: map[string]string{
+					convertFromCalicoAnnotation: "true",
+				},
+			},
+		}
+
+		if acnp.Spec.AppliedTo, err = convertSelectorToAntreaAppliedTo(gnp.Spec.Selector, gnp.Spec.NamespaceSelector); err != nil {
+			return err
+		}
+		if acnp.Spec.Ingress, err = convertCalicoRuleToAntreaRule(gnp.Spec.Ingress); err != nil {
+			return err
+		}
+		if acnp.Spec.Egress, err = convertCalicoRuleToAntreaRule(gnp.Spec.Egress); err != nil {
+			return err
+		}
+		if gnp.Spec.Order == nil {
+			acnp.Spec.Priority = defaultPriority
+		} else {
+			acnp.Spec.Priority = *gnp.Spec.Order
+		}
+
+		if _, err := c.antreaClient.CrdV1beta1().ClusterNetworkPolicies().Create(context.Background(), &acnp, metav1.CreateOptions{}); err != nil {
+			klog.Errorf("Failed to create ClusterNetworkPolicy %s: %v", acnp.Name, err)
+			return err
+		}
+	}
+	return nil
+}
+
+func convertCalicoRuleToAntreaRule(rule []v3.Rule) ([]v1beta1.Rule, error) {
+	var antreaRules []v1beta1.Rule
+
+	for _, r := range rule {
+		antreaRule := v1beta1.Rule{}
+		switch r.Action {
+		case v3.Allow:
+			antreaRule.Action = migrate.ToPtr(v1beta1.RuleActionAllow)
+		case v3.Log:
+			antreaRule.EnableLogging = true
+			antreaRule.Action = migrate.ToPtr(v1beta1.RuleActionAllow)
+		case v3.Deny:
+			antreaRule.Action = migrate.ToPtr(v1beta1.RuleActionReject)
+		case v3.Pass:
+			antreaRule.Action = migrate.ToPtr(v1beta1.RuleActionPass)
+		}
+		var err error
+		if antreaRule.From, err = convertEntityRuleToAntreaNetworkPolicyPeer(r.Source); err != nil {
+			return nil, err
+		}
+		if antreaRule.To, err = convertEntityRuleToAntreaNetworkPolicyPeer(r.Destination); err != nil {
+			return nil, err
+		}
+		if antreaRule.Ports, err = convertCalicoPortToAntreaPort(r); err != nil {
+			return nil, err
+		}
+		antreaRule.ToServices = convertCalicoServiceToAntreaService(r)
+		antreaRules = append(antreaRules, antreaRule)
+	}
+
+	if len(rule) != 0 {
+		antreaRules = append(antreaRules,
+			v1beta1.Rule{Action: migrate.ToPtr(v1beta1.RuleActionReject)})
```

Comment:
I don't get this, why Reject rule is added?

Reply:
In Calico NP, if one NP is applied to some Pods, then these Pods' ingress and egress traffic will be rejected except `Allow` fields in that NP, so I added a default Reject rule.